api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/SolrQueryTest.java

```diff
@@ -0,0 +1,96 @@
+
+import edu.cornell.mannlib.vitro.webapp.application.ApplicationImpl;
+import edu.cornell.mannlib.vitro.webapp.application.ApplicationUtils;
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.Parameter;
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.SolrQuery;
+import edu.cornell.mannlib.vitro.webapp.modules.Application;
+import edu.cornell.mannlib.vitro.webapp.modules.searchEngine.SearchEngine;
+import edu.cornell.mannlib.vitro.webapp.searchengine.base.BaseSearchQuery;
+import edu.cornell.mannlib.vitro.webapp.searchengine.solr.SolrSearchEngine;
+import org.easymock.Mock;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import stubs.javax.servlet.http.HttpServletRequestStub;
+
+@PowerMockIgnore("javax.management.*")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ApplicationUtils.class)
+public class SolrQueryTest {
+
+    public static SearchEngine mockSearchEngine;
+
+    private SolrQuery solrQuery;
+
+    @Mock
+    private Parameter parameter1;
+
+
+    @BeforeClass
+    public static void setupSolrMock(){
+
+    }
+
+    @Before
+    public void setupObjects() {
+        mockSearchEngine = createMock(SolrSearchEngine.class);
+
+        Application application =  createMock(ApplicationImpl.class);
+        expect(application.getSearchEngine()).andReturn(mockSearchEngine).anyTimes();
+        replay(application);
+
+        mockStatic(ApplicationUtils.class);
+        expect(ApplicationUtils.instance()).andReturn(application).anyTimes();
+        replay(ApplicationUtils.class);
+
+        this.solrQuery = new SolrQuery();
+    }
+
+    @Test
+    public void requiredParameterMissingFromInput(){
+        parameter1 = createMock(Parameter.class);
+        expect(parameter1.getName()).andReturn("testParameter").times(1);
+        replay(parameter1);
+        solrQuery.addRequiredParameter(parameter1);
+        assertTrue(solrQuery.run(new OperationData(new HttpServletRequestStub())).hasError());
+        verify(parameter1);
+    }
+
+    @Test
+    public void requiredParameterPresentButInvalid(){
+        parameter1 = createMock(Parameter.class);
+        expect(parameter1.getName()).andReturn("testParameter").times(1);
+        expect(parameter1.isValid("testParameter", new String[]{"testValue"}))
+            .andReturn(false).times(1);
+        replay(parameter1);
+        solrQuery.addRequiredParameter(parameter1);
+        HttpServletRequestStub httpReq = new HttpServletRequestStub();
+        httpReq.addParameter("testParameter","testValue");
+        assertTrue(solrQuery.run(new OperationData(httpReq)).hasError());
+        verify(parameter1);
+    }
+
+    @Test
+    @Ignore
```

Comment:
Is this still intended to be ignored?

Reply:
Yes, for now. Not sure how to evaluate returned value from SolrOperation until we decide what exactly will be returned within OperationResult. There are 5 more test scenarios I am planning on writing, but all require assertions on returned value, so I thought to do it once format of that returned value is agreed upon.

Reply:
I think it would be valuable to at least now the operation runs all the way through without exception for various inputs.